packages/trader/src/Modules/Contract/Containers/contract.tsx

```diff
@@ -1,16 +1,20 @@
-import PropTypes from 'prop-types';
 import React from 'react';
 import { Redirect, withRouter } from 'react-router';
 import { CSSTransition } from 'react-transition-group';
 import { routes } from '@deriv/shared';
 import ErrorComponent from 'App/Components/Elements/Errors';
 import { localize } from '@deriv/translations';
-import ContractReplay from './contract-replay.jsx';
+import ContractReplay from './contract-replay';
 import { observer, useStore } from '@deriv/stores';
 
+type TContract = {
```

Comment:
```suggestion
type TContractParams = { contract_id: string };
type TContract = RouteComponentProps<TContractParams>;
```

export interface RouteComponentProps<
    Params extends { [K in keyof Params]?: string } = {},
    C extends StaticContext = StaticContext,
    S = H.LocationState
> {
    history: H.History<S>;
    location: H.Location<S>;
    match: match<Params>;
    staticContext?: C | undefined;
}

Something like this. And go from there using typescript suggestions

Reply:
type TContractParams = { isExact: boolean; params: { contract_id: string }; path: string; url: string };

Could also be this. I'm not 100% sure.

Reply:
i'm not sure where you got `isExact, path and url` from

Reply:
if you apply this suggestion you don't have to type history.

Reply:
Will apply, thank you!
`isExact`, `path` and `url` were in console when I logged what I've got